coalib/results/result_actions/ShowAppliedPatchesAction.py

```diff
@@ -3,12 +3,7 @@
 from coalib.results.RESULT_SEVERITY import (
     RESULT_SEVERITY, RESULT_SEVERITY_COLORS)
 from pyprint.ConsolePrinter import ConsolePrinter
-
-
-def format_lines(lines, symbol='', line_nr=''):
-    def sym(x): return ']' if x is '[' else x
-    return '\n'.join('{}{:>5}{} {}'.format(symbol, sym(symbol), line_nr, line)
-                     for line in lines.rstrip('\n').split('\n'))
+from coalib.output.Interactions import format_lines
```

Comment:
why do we even need to use format_lines in this module?

And are you sure that padding of 5 is not intentional, and 4 is good enough?

Reply:
> And are you sure that padding of 5 is not intentional, and 4 is good enough?

In ShowAppliedPatchesAction, `format_lines()` is used when [printing the `Severity`](https://github.com/coala/coala/blob/master/coalib/results/result_actions/ShowAppliedPatchesAction.py#L45) of the patch.
Here's how it looked (before this PR):
*Notice the line* `!    ![Severity: NORMAL]`
![showappliedpatchaction_on_master](https://user-images.githubusercontent.com/24698858/38816832-9a8b0dd8-41b4-11e8-8b69-e424f0f35065.png)
And here's how it looks now (after this PR):
![showappliedpatchesaction_on_removeredun3](https://user-images.githubusercontent.com/24698858/38816917-ce7e7f3a-41b4-11e8-86ed-7b6cf1517bec.png)
So, the version of `format_lines()` in this PR works like it was expected to IMO. It also passes all the original set of tests as they were, without any modification.

> why do we even need to use format_lines in this module?

Yes I agree, that was a bad decision on my part. :sweat_smile:
I wasn't sure if creating a new file to solve a *difficulty/low* issue was allowed or not. Now I think it would be better if I create a new file (`Formatting.py` maybe?) and put `format_lines()`, `color_letter()` and `highlight_text()` inside it. What do you think @jayvdb ?

Reply:
> what does format_lines do?

`format_lines` returns a formatted version of the string passed to it as a parameter. It formats the string to make it look elegantly in coala's CLI by adding a little prefix to it. 

> Why is that functionality needed in ShowAppliedPatchesAction.py?

`ShowAppliedPatchesAction` already uses it to format the line which displays the Severity of any applied action.

So, unless you are suggesting that we change how `ShowAppliedPatchesAction` shows its messages, I think that `format_lines` is the right tool for the job.

Reply:
You still have not read `format_lines` and its usages carefully.
It is gobbledy-gook that people have copied and pasted without thinking.

Reply:
> You still have not read format_lines and its usages carefully.

:sweat_smile: I'm sorry, I didn't find anything else to read about it except the 4 line definition which was present in all 3 of the above mentioned files.

So, are you suggesting that we modify the definition of `format_lines()`? maybe make pass the amount of padding as a parameter? Or are you suggesting that we remove the function altogether?

Reply:
Read those four lines very carefully.  Understand exactly what happens when it is called each time.

Reply:
Here's what I understand `format_lines()` is doing:
Each time the function is called with a string (`lines`), it splits that string at `\n`s, prefixes each split string with the `symbol` (passed to it as a parameter), adds a padding of 4, adds the `sym()` version of `symbol`, adds `line_nr` and finally joins them using `\n` and returns this joined version.
Also, `sym()` is a local function inside `format_lines()` which is just responsible to return the closing `]` in case the `symbol` that is passed is `[`. Otherwise, it returns the passed character itself.

Isn't that all?  :sweat_smile:

Reply:
No I dont want you to tell me what it does in theory.  I already know the answers.

You need to see why in practise , in some of the real world scenarios where it is used, it isnt particularly useful.